### PR TITLE
profiler: Add batch ebpf map operations (reloaded)

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

### CODE REVIEW

**Comment by kakkoyun** [comment]
cc @javierhonduco @v-thakkar

File: pkg/profiler/profiler.go [hunk 1]
```diff
@@ -417,48 +390,87 @@ func (p *CgroupProfiler) profileLoop(ctx context.Context) error {
...
+
+	if processedCount == 0 {
+		level.Error(p.logger).Log("msg", "no values in batch")
+		return nil
+	}
+
+	// We are getting and deleting the whole map, so there should not be a next batch.
+	if nextCountKey != uintptr(0) {
+		level.Debug(p.logger).Log("msg", "Next batch should be null", "nextBatch", nextCountKey)
+	}
```
Comment (v-thakkar) [inline pkg/profiler/profiler.go:426]: I like that we're checking this :tada:

**Comment by v-thakkar** [comment]
LGTM!

**Review by javierhonduco — APPROVED** [review]
LGTM! 😄

Do we need to update the minikube ISO somewhere or it's automatically updated? Before merging, could you run an e2e test with the cpp binary if you haven't already?

**Comment by kakkoyun** [comment]
> Do we need to update the minikube ISO somewhere or it's automatically updated?

We only document it and don't actually force any minikube iso version (except e2e tests cc @Sylfrena). I have already updated the documentation in repo and I have a PR to update parca.dev

> Before merging, could you run an e2e test with the cpp binary if you haven't already?

Will do :+1:

**Comment by javierhonduco** [comment]
Was thinking of batch BPF operations in the context of cleaning up maps. Let's not forget to take a look at the profiles and see if it could be worth it to batch-delete our maps 😄

**Comment by javierhonduco** [comment]
Some quick data (https://pprof.me/3f07325):

- `pkg/profiler/cpu.(*bpfMaps).clean`: 2.8% of the cycles
- `github.com/aquasecurity/libbpfgo.(*BPFMapIterator).Next` 0.9% of the cycles each
- `pkg/profiler/cpu.(*bpfMaps).readKernelStack`, `readKernelStack`, `readUserStackWithDwarf` : 0.9% of the cycles each

**Pull request closed without merging**